### ISSUE TITLE
Demuxlet FATAL ERROR -  Cannot extract R2 (1 float value) from INFO field. Cannot use --geno-error-coeff

### ISSUE THREAD

**Issue by changostraw** [drneavin/Demultiplexing_Doublet_Detecting_Docs#17]
Hi Drew,
After running Demuxlet successfully on all my test data, I sequenced my real cohort and have started to demultiplex my real cohort data. The only difference between the files is that the real cohort data contains 3x more nuclei, so the bam files are a lot larger. Due to this I am now having issues running Demuxlet. 
1. the popscle pileup takes days and days to run and has no checkpoint so if it times out or runs out of memory I have to start all over at the beginning. Is there a way to add a checkpoint option so that I can restart it from where it stopped?
2. When pileup is successful, I get an error when running demuxlet that I haven't seen before: < FATAL ERROR - 
Cannot extract R2 (1 float value) from INFO field at chr1:817341. Cannot use --geno-error-coeff. >
I am not sure what this means. Is it a problem with my vcf file? I have tried both raw genotype vcf and the imputed genotype vcf. The files are the same as when I ran my test file. 
Thanks!

**Comment by changostraw** [comment]
oh, and one more question:
3. is there an option to increase threads for demuxlet and freemuxlet? That seems to really help running souporcell as I can increase the threads to 33. 
Thanks!

**Comment by drneavin** [comment]
Hi @changostraw,

Exciting that you are able to work on your cohort data now! A couple of questions to start:

1.  How many nuclei are in your barcode file and how many snps do you have in your snp genotype file after filtering for snps overlapping genes?

To address your questions:
1. Popscle hasn't built in the type of checkpoints you're talking about so unfortunately, restarting from the beginning is the only viable option currently
2. Can you check if you have `R2` in the header of your vcf and also in the FORMAT field of your vcf? That error suggests that `R2` is not a parameter that is available in the vcf. Depending on your imputation method, another metric often used for imputation confidence is `INFO` (not to be confused with the `INFO` vcf column)
3. Unfortunately, popscle doesn't have an option to indicate the number of threads but I am unsure whether it may internally assess the number of available threads/cores and use them.

Cheers,
Drew

**Comment by changostraw** [comment]
Thanks @drneavin!
I don't have an R2 column, but I do have a FORMAT column in my vcf:
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	10_GT00268	111_GT02088	114_GT02170... and more samples etc.

It's strange because I didn't get this error before during the test samples and I was using the same vcf file. I did update the demuxafy image though, so perhaps something changed with a newer version of demuxlet?

**Comment by changostraw** [comment]
Also, for the raw genotype vcf, imputation quality is not applicable, so can I cancel the --geno-error-coeff option?

**Comment by changostraw** [comment]
Hi again,
I solved the issue with raw genotype vcf. I just set both --geno-error-coeff  --geno-error-offset  to zero as they shouldn't be needed for non imputed genotypes and it ran without error. 

I also realized I was trying to use r2-info INFO option on pileup instead of demuxlet. Now I am using it on demuxlet but getting the same error that it can't find INFO (instead of R2). 

FATAL ERROR - 
Cannot extract INFO (1 float value) from INFO field at chr1:817341. Cannot use --geno-error-coeff

In the INFO column the vcf has the values: 
INFO=1.000;AC=7;AN=8
INFO=0.996;AC=1;AN=8
INFO=1.000;AC=1;AN=8

Perhaps I did something wrong with the vcf file quality scores?
thanks!
